language/move-prover/spec-lang/src/ast.rs

```diff
@@ -60,6 +65,8 @@ pub enum ConditionKind {
     Ensures,
     Requires,
     RequiresModule,
+    RequiresSpecCheck,
```

Comment:
Do we really need a new condition kind for this? This will leak all over the place where we are dealing with condition kinds. If it's needed what do these actually mean? Is it general or specific for one of those tests? In the later case, it wouldn't scale to introduce a new kind for each of those checks.